picard/file.py

```diff
@@ -464,6 +467,12 @@ def _acoustid_update(self):
             recording_id = self.metadata['musicbrainz_recordingid']
         self.tagger.acoustidmanager.update(self, recording_id)
 
+    def analyze(self, use_acoustid):
+        self.set_pending()
+        if use_acoustid:
+            callback = partial(self._lookup_finished, File.LOOKUP_ACOUSTID)
+            self.tagger.acoustidclient.analyze(self, callback)
```

Comment:
While in theory I liked the idea of making this a method of `File`, I don't like the use of `self.tagger.acoustidmanager` here. IMHO we should reduce the access of properties of the tagger object. In this sense this code makes more sense inside tagger itself.

Side question: What happens to set_pending if `use_acoustid == False`? Does the pending state get reset?

Reply:
While i agree about the general idea to reduce the access of properties of the tagger object, that quite unrealistic with the current code, here are the number of occurences of `self.tagger.` among source files in master:

```
     36 picard/ui/mainwindow.py
     23 picard/file.py
     18 picard/album.py
     12 picard/ui/itemviews.py
     12 picard/cluster.py
     11 picard/ui/searchdialog/track.py
      9 picard/ui/searchdialog/album.py
      7 picard/acoustid/__init__.py
      5 picard/ui/ui_options_script.py
      5 picard/ui/options/general.py
      5 picard/browser/browser.py
      5 picard/acoustid/manager.py
      4 picard/collection.py
      3 picard/ui/options/network.py
      3 picard/ui/coverartbox.py
      3 picard/disc.py
      2 picard/ui/searchdialog/artist.py
      2 picard/ui/scriptsmenu.py
      2 picard/ui/options/tags.py
      2 picard/ui/options/renaming.py
      2 picard/ui/options/plugins.py
      2 picard/ui/options/metadata.py
      2 picard/ui/cdlookup.py
      2 picard/track.py
      2 picard/releasegroup.py
      2 picard/plugin.py
      1 test/picardtestcase.py
      1 picard/webservice/__init__.py
      1 picard/util/checkupdate.py
      1 picard/ui/util.py
      1 picard/ui/searchdialog/__init__.py
      1 picard/ui/ratingwidget.py
      1 picard/ui/options/releases.py
      1 picard/ui/options/interface.py
      1 picard/ui/metadatabox.py
```

Imho File class needs access to acoustid stuff (after all, acoustid applies to files).
But i agree this move isn't the best, we need to improve that somehow.

Do you have ideas about how we could improve the situation ??

> Side question: What happens to set_pending if `use_acoustid == False`? Does the pending state get reset?

Good question, it doesn't seem so, but the code is damn messy.

Reply:
> Side question: What happens to set_pending if `use_acoustid == False`? Does the pending state get reset?

Imho the original code is buggy, what's the point of calling `set_pending()` if nothing is called behind ?

Reply:
The pending files issue was addressed by c948e014004806d9d7a24c1a519d734aaef7a9b1